### PR TITLE
Fixed web-bluetooth types as filters are optionals

### PR DESCRIPTION
**Pull request by xlozinguez**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:  on this article https://developers.google.com/web/updates/2015/07/interact-with-ble-devices-on-the-web
```
Finally, *instead of filters you can use the acceptAllDevices* key to show all nearby Bluetooth devices. You will also need to define the optionalServices key to be able to access some services. If you don't, you'll get an error later when trying to access them.
```

### CODE REVIEW

**Comment by urish** [comment]
This allows `requestDevices({})`, which according [to the spec](https://webbluetoothcg.github.io/web-bluetooth/#example-disallowed-filters) is invalid. Seems like either `filters` or `acceptAllDevices` can be specified, but [not both](https://webbluetoothcg.github.io/web-bluetooth/#example-disallowed-filters). 

Can you please update the PR to reflect that?

**Comment by xlozinguez** [comment]
@urish agreed, I've looked into the [advanced types](https://www.typescriptlang.org/docs/handbook/advanced-types.html) docs and I tried the following 

```typescript
type RequestDeviceOptions = {
	filters: BluetoothRequestDeviceFilter[];
	optionalServices?: BluetoothServiceUUID[];
} | {
	acceptAllDevices: boolean;
	optionalServices?: BluetoothServiceUUID[];
};
```

For some reason, however, the linter wouldn't bulge if you were to specify both `filters` and `acceptAllDevices` params... Do you know why that would be? See [this link](https://www.typescriptlang.org/play/index.html#src=type%20foo%20%3D%20%7B%0D%0A%20%20%20%20name%3A%20string%2C%0D%0A%20%20%20%20loved%3F%3A%20boolean%0D%0A%7D%20%7C%20%7B%0D%0A%20%20%20%20surname%3A%20string%2C%0D%0A%20%20%20%20loved%3F%3A%20boolean%0D%0A%20%20%20%20%7D%0D%0A%0D%0Afunction%20fiz(test%3A%20foo)%20%7B%0D%0A%20%20%20%20console.log(test)%3B%0D%0A%7D%0D%0A%0D%0Afiz%20(%7B%0D%0A%20%20%20%20name%3A%20'bob'%2C%0D%0A%20%20%20%20surname%3A%20'dylan'%0D%0A%7D)%3B).

**Comment by urish** [comment]
Yes, unfortunately, this is not currently possible with TypeScript, see the discussion in Microsoft/TypeScript#14094.

Still, this approach makes sure that at least `filters` or `acceptAllDevices` are specified...

Also, I believe that specifying both `filters` and `acceptAllDevices: false` is also valid, so perhaps it is even better with the Union type.

**Comment by xlozinguez** [comment]
I literally google'ed my way to the same exact discussion! Great, I push that change through in a minute.

**Comment by xlozinguez** [comment]
Just pushed the update @urish if you don't mind

File: types/web-bluetooth/index.d.ts [hunk 1]
```diff
@@ -98,11 +101,9 @@ interface BluetoothDevice extends EventTarget, BluetoothDeviceEventHandlers, Cha
 	readonly name?: string;
 	readonly gatt?: BluetoothRemoteGATTServer;
 	readonly uuids?: string[];
-
```
Comment (urish) [inline types/web-bluetooth/index.d.ts]: any special reason for removing the blank lines?

Comment (xlozinguez) [reply types/web-bluetooth/index.d.ts]: Not other than to be consistent with the other part of the files. Happy to put them back in :)

**Comment by urish** [comment]
👍

**Comment by manekinekko** [comment]
Awesome! Great job @xlozinguez. Thanks, @urish for approving this PR.

**Comment by manekinekko** [comment]
@xlozinguez can you please ping me when this PR is merged?

**Pull request merged**